Math/NumberTheory/Diophantine.hs

```diff
@@ -64,3 +75,101 @@ cornacchia d m
  where
   candidates = map (\sf -> (sf, m `div` (sf * sf))) (squareFactors m)
   solve (sf, m') = map (\(x, y) -> (x * sf, y * sf)) (cornacchiaPrimitive d m')
+
+----
+
+-- | A linear diophantine equation `ax + by = c`
+-- | where `x` and `y` are unknown
+data Linear a = Lin { a,b,c :: a }
+  deriving
+    ( Show, Eq, Ord
+    )
+
+-- | A solution to a linear equation
+data LinearSolution a = LS { x,v,y,u :: a }
+  deriving
+    ( Show, Eq, Ord
+    )
+
+-- | Produces an unique solution given any
+-- | arbitrary number k
+runLinearSolution :: _ => LinearSolution b -> b -> (b, b)
```

Comment:
Could you please add tests, showing that 
a) everything this function returns **is** a solution,
b) all possible solutions are generated by this function by supplying different `k`?

Reply:
There are either no solutions, or an infinite number of solutions. I'll add some test cases, but I cannot check it exhaustively